### ISSUE TITLE
Error when creating custom Quill instance with List

### ISSUE THREAD

**Issue by ringvold** [quilljs/quill#1383]
When building a custom Quill instance by using Quill core and registrating selected formats I get a Parchement error: `Uncaught Error: [Parchment] Unable to create list-item blot`.
Does not occure when using prebuild quill.js.

Have tried it with and without also importing and registrating ListItem.

Am I just doing it wrong?
https://github.com/ringvold/webpack-example/blob/ab1de33151a8a13da28e4f494fbfb9e5f2279fc3/app.js

**Steps for Reproduction**

Example with prebuild bundle: 
1. Visit http://codepen.io/ringvold/pen/wJYNmz
2. Open console (browser developer tools, not codepen console)
3. Try to make a list in the editor
4. Observe error in console

Example with building from source:
1. Clone git@github.com:ringvold/webpack-example.git`
2. Yarn/npm install
3. Yarn/npm run build
4. Open index.html in browser
5. Open console (browser developer tools, not codepen console)
6. Try to make a list in the editor
7. Observe error in console

**Expected behavior**:
Creating lists should work when importing and registrating List with Quill.

**Actual behavior**:
Error `Uncaught Error: [Parchment] Unable to create list-item blot` when trying to create a list in the editor.

**Platforms**:
Tested on Windows 7 and OSX Sierra in Chrome version 57.0.2987.110.
See yarn.lock in https://github.com/ringvold/webpack-example for information on packages used for bulding bundle.

**Version**:
`Quill.version` say dev as it is a custom build, but it is based on version 1.2.3.

**Comment by ringvold** [comment]
Never mind. It seems that I am doing it wrong. -_- 

After looking though the quill source I found that ListItem must be registered as `formats/list/item`, _not_ `formats/list-item`. I found this in main module, [quill.js](https://github.com/quilljs/quill/blob/develop/quill.js).
Also it seems it must be imported as `import List, { ListItem } from 'quill/formats/list'`, but that is as ESs6 detail I guess.

It seems wierd that List do not work without also registration ListItem though.

The registration paths in Quill can be quite confusing, espesially with custom build of quill where the formats source location do not match the registration path.  Maybe that could be more clear in the documentation. I see that the quill and core modules are linked to in [Adding Quill to Your Build Pipeline](http://quilljs.com/guides/adding-quill-to-your-build-pipeline/) but maybe there could be a section on registration paths or highlight the importance of the registration path?
